internal/controller/apiextensions/composite/composition_ptf.go

```diff
@@ -724,10 +730,34 @@ func RunFunction(ctx context.Context, fnio *iov1alpha1.FunctionIO, fn *v1.Contai
 		return nil, errors.Wrap(err, errDialRunner)
 	}
 
+	k8schainOpts := k8schain.Options{}
+	if sa := os.Getenv("POD_SERVICE_ACCOUNT"); sa != "" {
+		k8schainOpts.ServiceAccountName = sa
+	}
+	if ns := os.Getenv("POD_NAMESPACE"); ns != "" {
+		k8schainOpts.Namespace = ns
+	}
```

Comment:
We usually avoid reading environment values in arbitrary places in the codebase and, instead, parse them in all in a single location during startup and flow them down with options. 

The two variables here are already [parsed](https://github.com/crossplane/crossplane/blob/580d15ac8ac79e72acec7f95a70c8ddd9026d6b8/cmd/crossplane/core/core.go#L70-L71), and we should better pass them down [similar to how the package manager does](https://github.com/crossplane/crossplane/blob/580d15ac8ac79e72acec7f95a70c8ddd9026d6b8/cmd/crossplane/core/core.go#L171-L172).